[Automation] native-image run fails for org.flywaydb:flyway-sqlserver:12.3.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27055853375
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.flywaydb.flyway-sqlserver_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.12.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-databind/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-core/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=30a2b2d3-91e4-e685-119a-78977294d2ea \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.9s @ 0.17GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (56.1s @ 1.28GB)
   13,582 types,  18,332 fields, and  65,788 methods found reachable
    4,459 types,     713 fields, and   5,001 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (7.5s @ 1.49GB)
[4/8] Parsing methods...      [***]                                                                      (5.4s @ 1.62GB)
[5/8] Inlining methods...     [***]                                                                      (3.1s @ 1.72GB)
[6/8] Compiling methods...    [*****]                                                                   (30.6s @ 1.21GB)
[7/8] Laying out methods...   [***]                                                                      (5.3s @ 1.57GB)
[8/8] Creating image...       [***]                                                                      (5.4s @ 1.76GB)
  22.10MB (38.63%) for code area:    42,365 compilation units
  28.64MB (50.06%) for image heap:  345,463 objects and 168 resources
   6.48MB (11.32%) for other data
  57.21MB in total image size, 53.55MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   8.99MB java.base                                            6.54MB byte[] for code metadata
   3.65MB svm.jar (Native Image)                               4.74MB byte[] for java.lang.String
   2.66MB java.xml                                             4.61MB char[]
   1.44MB mssql-jdbc-12.2.0.jre11.jar                          2.22MB java.lang.String
   1.23MB jackson-databind-3.1.0.jar                           1.78MB java.lang.Class
 628.92kB flyway-core-12.3.0.jar                               1.41MB com.oracle.svm.core.hub.DynamicHubCompanion
 556.65kB assertj-core-3.22.0.jar                              1.29MB byte[] for general heap data
 317.40kB jackson-core-3.1.0.jar                             478.08kB java.util.concurrent.ConcurrentHashMap$Node
 236.48kB java.rmi                                           472.08kB byte[] for reflection metadata
 198.63kB junit-jupiter-engine-5.9.2.jar                     441.82kB java.util.HashMap$Node
   1.52MB for 40 more packages                                 4.64MB for 3465 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 27 component(s) is embedded in binary (1.33kB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       10.4s (8.4% of total time) in 1224 GCs | Peak RSS: 2.66GB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 2m 3s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Starting MSSQL ...
Jun 06, 2026 8:23:25 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 4a276e17-f785-4964-b528-e017d4730e6c Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:4a276e17-f785-4964-b528-e017d4730e6c
Jun 06, 2026 8:23:25 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: e8edb6bf-d114-4f88-b969-d73245669acc Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:e8edb6bf-d114-4f88-b969-d73245669acc
Jun 06, 2026 8:23:25 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 061da966-0e80-4338-92f0-559d2e882f32 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:061da966-0e80-4338-92f0-559d2e882f32
Jun 06, 2026 8:23:26 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 12f593cf-0da6-4a71-b23f-4fe578adc5f1 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:12f593cf-0da6-4a71-b23f-4fe578adc5f1
Jun 06, 2026 8:23:27 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 14a319e2-6492-4a83-a550-805edf58be71 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:14a319e2-6492-4a83-a550-805edf58be71
Jun 06, 2026 8:23:28 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: e0361a53-f861-4a69-a144-8960630d59f5 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:e0361a53-f861-4a69-a144-8960630d59f5

FAILURE: Build failed with an exception.

MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.flywaydb.core.Flyway.<init>(Flyway.java:138)
       flyway.sqlserver.FlywaySqlServerTests.migrate(FlywaySqlServerTests.java:73)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
     Caused by: org.flywaydb.core.api.FlywayException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromSlug(FlywayDbWebsiteLinks.java:102)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromTopic(FlywayDbWebsiteLinks.java:107)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.helpUrl(FlywayDbWebsiteLinks.java:89)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.<clinit>(FlywayDbWebsiteLinks.java:36)
       [...]
     Caused by: java.net.MalformedURLException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.unsupported(JavaNetSubstitutions.java:247)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.getURLStreamHandler(JavaNetSubstitutions.java:221)
       java.base@25.0.3/java.net.URL.getURLStreamHandler(URL.java:80)
       java.base@25.0.3/java.net.URL.<init>(URL.java:741)
       java.base@25.0.3/java.net.URL.of(URL.java:871)
       [...]

Test run finished after 4632 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html
* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 17s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.12.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 19s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 12.3.0 at 2026-06-06T08:23:29Z with exit code 1 after 139s
❌ ERROR: 'native-image run' for 12.3.0 failed with exit code 1.
FAILED[native-image run][12.3.0][env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27055853375
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ork/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.12.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-databind/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-core/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=30a2b2d3-91e4-e685-119a-78977294d2ea \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.8s @ 0.24GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (55.4s @ 1.19GiB)
   13,583 types,  18,390 fields, and  66,070 methods found reachable
      579 types,     115 fields, and   1,965 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
      168 resource accesses registered with 322.32KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.9s @ 1.35GiB)
[4/8] Parsing methods...      [***]                                                                     (5.0s @ 1.48GiB)
[5/8] Inlining methods...     [***]                                                                     (3.0s @ 1.57GiB)
[6/8] Compiling methods...    [******]                                                                 (33.1s @ 1.14GiB)
[7/8] Laying out methods...   [***]                                                                     (6.0s @ 1.48GiB)
[8/8] Creating image...       [***]                                                                     (5.3s @ 1.77GiB)
  19.03MiB (37.43%) for code area:    42,461 compilation units
  27.69MiB (54.47%) for image heap:  359,750 objects and 32 resources
   4.12MiB ( 8.10%) for other data
  50.83MiB in total image size, 47.31MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.85MiB o.g.n.runtime/com.oracle.svm.core.code              5.67MiB byte[] for code metadata
   1.15MiB com.microsoft.sqlserver.jdbc                        4.59MiB byte[] for string data
 860.34KiB java.base/java.lang.invoke                          4.40MiB java.base/char[]
 848.55KiB java.base/sun.security.ssl                          2.18MiB java.base/java.lang.String
 774.80KiB java.base/java.util                                 1.70MiB java.base/java.lang.Class
 465.15KiB java.base/com.sun.crypto.provider                   1.45MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 407.16KiB java.base/jdk.internal.classfile.impl               1.14MiB byte[] for general heap data
 390.12KiB java.base/java.lang                               585.03KiB byte[] for reflection metadata
 306.94KiB java.xml/com.sun.org.apache.xerces.internal.impl  526.48KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 306.22KiB java.base/java.util.concurrent                    514.34KiB heap alignment
  11.77MiB for 495 more packages                               4.97MiB for 3494 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.28KiB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                      10.7s (8.4% of total time) in 1293 GCs | Peak RSS: 2.53GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable, 47.31MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 2m 5s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Starting MSSQL ...
Jun 06, 2026 8:24:28 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: f8f3d7ea-33cf-4c2b-9f0d-c11e0c9af6a3 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:f8f3d7ea-33cf-4c2b-9f0d-c11e0c9af6a3
Jun 06, 2026 8:24:28 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 1e4adae4-8748-4470-9392-987bb716aeca Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:1e4adae4-8748-4470-9392-987bb716aeca
Jun 06, 2026 8:24:29 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: a860bf00-63a4-4cc0-92c6-eac1f58994ed Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:a860bf00-63a4-4cc0-92c6-eac1f58994ed
Jun 06, 2026 8:24:29 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 31f3fabf-4d2a-461b-a340-8be107d9e608 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:31f3fabf-4d2a-461b-a340-8be107d9e608
Jun 06, 2026 8:24:30 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 8c14cc63-dcca-4315-aa81-82137679e8b6 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:8c14cc63-dcca-4315-aa81-82137679e8b6
Jun 06, 2026 8:24:31 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: fdaad5c5-440d-4d13-a6e2-415e2e9cceeb Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:fdaad5c5-440d-4d13-a6e2-415e2e9cceeb
Jun 06, 2026 8:24:32 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: cb862684-a482-4425-b6c6-2135aec99084 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:cb862684-a482-4425-b6c6-2135aec99084
MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.flywaydb.core.Flyway.<init>(Flyway.java:138)
       flyway.sqlserver.FlywaySqlServerTests.migrate(FlywaySqlServerTests.java:73)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
     Caused by: org.flywaydb.core.api.FlywayException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromSlug(FlywayDbWebsiteLinks.java:102)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromTopic(FlywayDbWebsiteLinks.java:107)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.helpUrl(FlywayDbWebsiteLinks.java:89)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.<clinit>(FlywayDbWebsiteLinks.java:36)
       [...]
     Caused by: java.net.MalformedURLException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.unsupported(JavaNetSubstitutions.java:426)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.getURLStreamHandler(JavaNetSubstitutions.java:400)
       java.base@25.0.3/java.net.URL.getURLStreamHandler(URL.java:96)
       java.base@25.0.3/java.net.URL.<init>(URL.java:741)
       java.base@25.0.3/java.net.URL.of(URL.java:871)
       [...]

Test run finished after 5632 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 20s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.12.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 22s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 12.3.0 at 2026-06-06T08:24:34Z with exit code 1 after 143s
❌ ERROR: 'native-image run' for 12.3.0 failed with exit code 1.
FAILED[native-image run][12.3.0][env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27055853375
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
form-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.12.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-databind/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/tools.jackson.core/jackson-core/3.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=71c8517f-d29b-5099-b0ad-9bf4f4397f22 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.7s @ 0.25GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (56.9s @ 1.23GiB)
   13,836 types,  18,866 fields, and  67,597 methods found reachable
      597 types,     115 fields, and   1,984 methods registered for reflection
       64 types,      73 fields, and      55 methods registered for JNI access
      170 resource accesses registered with 322.46KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.5s @ 1.43GiB)
[4/8] Parsing methods...      [***]                                                                     (6.0s @ 1.53GiB)
[5/8] Inlining methods...     [***]                                                                     (3.3s @ 1.66GiB)
[6/8] Compiling methods...    [******]                                                                 (35.5s @ 1.32GiB)
[7/8] Laying out methods...   [***]                                                                     (5.5s @ 1.69GiB)
[8/8] Creating image...       [***]                                                                     (5.1s @ 1.86GiB)
  19.52MiB (37.30%) for code area:    43,573 compilation units
  28.38MiB (54.23%) for image heap:  378,827 objects and 34 resources
   4.43MiB ( 8.47%) for other data
  52.32MiB in total image size, 48.69MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.89MiB o.g.n.runtime/com.oracle.svm.core.code              5.83MiB byte[] for code metadata
   1.15MiB com.microsoft.sqlserver.jdbc                        4.80MiB byte[] for string data
 864.21KiB java.base/sun.security.ssl                          4.40MiB java.base/char[]
 860.34KiB java.base/java.lang.invoke                          2.26MiB java.base/java.lang.String
 777.84KiB java.base/java.util                                 1.74MiB java.base/java.lang.Class
 464.51KiB java.base/com.sun.crypto.provider                   1.48MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 407.12KiB java.base/jdk.internal.classfile.impl               1.14MiB byte[] for general heap data
 401.56KiB java.base/java.lang                               605.20KiB byte[] for reflection metadata
 307.64KiB java.base/java.util.concurrent                    536.37KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 306.94KiB java.xml/com.sun.org.apache.xerces.internal.impl  444.31KiB java.base/java.util.HashMap$Node
  12.19MiB for 500 more packages                               5.18MiB for 3551 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.28KiB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                      10.4s (7.9% of total time) in 1356 GCs | Peak RSS: 2.59GiB | CPU load: 3.66
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable, 48.69MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 2m 9s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile
Jun 06, 2026 8:29:54 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 391d4970-0f43-4296-bafb-dbdeb67efded Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 8:29:54 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: a24c4301-766b-45d4-8268-2554b3a08eea Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:a24c4301-766b-45d4-8268-2554b3a08eea
Jun 06, 2026 8:29:54 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 5cdeb539-a007-48b6-b5eb-3fe26574eb31 Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 8:29:55 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 09eacd7f-83d8-43d1-8632-718d73a22012 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:09eacd7f-83d8-43d1-8632-718d73a22012

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Starting MSSQL ...
Jun 06, 2026 8:29:56 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 561ae5c6-1e9c-4247-a6a4-edd4dd2c879e Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 8:29:57 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 43fbb132-8473-427d-badf-2e4487acf866 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:43fbb132-8473-427d-badf-2e4487acf866
MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.flywaydb.core.Flyway.<init>(Flyway.java:138)
       flyway.sqlserver.FlywaySqlServerTests.migrate(FlywaySqlServerTests.java:73)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
     Caused by: org.flywaydb.core.api.FlywayException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromSlug(FlywayDbWebsiteLinks.java:102)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.getRedirectLinkFromTopic(FlywayDbWebsiteLinks.java:107)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.helpUrl(FlywayDbWebsiteLinks.java:89)
       org.flywaydb.core.internal.util.FlywayDbWebsiteLinks.<clinit>(FlywayDbWebsiteLinks.java:36)
       [...]
     Caused by: java.net.MalformedURLException: Accessing a URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.unsupported(JavaNetSubstitutions.java:426)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JavaNetSubstitutions.getURLStreamHandler(JavaNetSubstitutions.java:400)
       java.base@25.0.3/java.net.URL.getURLStreamHandler(URL.java:96)
       java.base@25.0.3/java.net.URL.<init>(URL.java:741)
       java.base@25.0.3/java.net.URL.of(URL.java:871)
       [...]

Test run finished after 4633 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 23s

> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.12.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 25s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 12.3.0 at 2026-06-06T08:29:58Z with exit code 1 after 145s
❌ ERROR: 'native-image run' for 12.3.0 failed with exit code 1.
FAILED[native-image run][12.3.0][env GVM_TCK_LV=12.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.12.0]
```

--- a/metadata/org.flywaydb/flyway-sqlserver/12.3.0/reachability-metadata.json
+++ b/metadata/org.flywaydb/flyway-sqlserver/12.3.0/reachability-metadata.json
@@ -1,0 +1,675 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.GaloisCounterMode$AESGCM",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsKeyMaterialGenerator",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsMasterSecretGenerator",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsPrfGenerator$V12",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "java.lang.Comparable"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "java.lang.String[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "java.net.Socket",
+      "methods" : [
+        {
+          "name" : "setOption",
+          "parameterTypes" : [
+            "java.net.SocketOption",
+            "java.lang.Object"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.apache.commons.logging.Log"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.apache.logging.log4j.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.clean.CleanModeCommandExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.clean.database.SQLServerCleanModePlugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineAppliedMigration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getBaselineMigrationPrefix",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setBaselineMigrationPrefix",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineResourceTypeProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.extensibility.Plugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.NullFlywayTelemetryManager"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.CleanModel"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.SchemaModel"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.EnvironmentProvisionerNone"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.EnvironmentVariableResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.PlaceholderPropertyResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.base.TestContainersDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.h2.H2DatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.sqlite.SQLiteDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.jdbc.ErrorOverrideInitializerStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.AuthCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.CheckCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DeployCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DiffCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DiffTextCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.GenerateCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.LicensingConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.LicensingConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.ModelCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.OfflinePermitConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.OfflinePermitConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PATTokenConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PATTokenConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PrepareCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.UndoCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isCheckDriftOnMigrate",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isPublishResult",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setCheckDriftOnMigrate",
+          "parameterTypes" : [
+            "boolean"
+          ]
+        },
+        {
+          "name" : "setPublishResult",
+          "parameterTypes" : [
+            "boolean"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.resource.CoreResourceTypeProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.scanner.classpath.ClasspathLocationHandlerImpl"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.scanner.filesystem.FilesystemLocationHandler"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.schemahistory.BaseAppliedMigration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.KerberosModel",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getLogin",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setLogin",
+          "parameterTypes" : [
+            "org.flywaydb.database.sqlserver.LoginModel"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.LoginModel",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getFile",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setFile",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getClean",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getKerberos",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setClean",
+          "parameterTypes" : [
+            "org.flywaydb.core.internal.command.clean.CleanModel"
+          ]
+        },
+        {
+          "name" : "setKerberos",
+          "parameterTypes" : [
+            "org.flywaydb.database.sqlserver.KerberosModel"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.babelfish.BabelfishDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.fabricDataWarehouse.FabricDataWarehouseDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.synapse.SynapseDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.slf4j.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.ClassUtils"
+      },
+      "type" : "org.slf4j.spi.SLF4JServiceProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.NativePRNG",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.security.SecureRandomParameters"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcTemplate"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.database.sqlserver.SQLServerConnection"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.database.sqlserver.synapse.SynapseDatabaseType"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.X509Factory",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.rsa.RSAPSSSignature",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.ssl.SSLContextImpl$TLSContext",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "tools.jackson.databind.deser.Deserializers[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "tools.jackson.databind.deser.KeyDeserializers[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "tools.jackson.databind.deser.ValueInstantiators[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "tools.jackson.databind.ser.Serializers[]"
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FlywayDbWebsiteLinks"
+      },
+      "glob" : "META-INF/services/java.net.spi.URLStreamHandlerProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "glob" : "META-INF/services/org.flywaydb.core.extensibility.Plugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.scanner.ClasspathClassScanner"
+      },
+      "glob" : "db/callback"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.scanner.classpath.ClasspathLocationHandlerImpl"
+      },
+      "glob" : "db/migration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.license.VersionPrinter"
+      },
+      "glob" : "org/flywaydb/core/internal/version.txt"
+    },
+    {
+      "bundle" : "com.microsoft.sqlserver.jdbc.SQLServerResource"
+    },
+    {
+      "bundle" : "sun.util.logging.resources.logging"
+    }
+  ]
+}

--- a/metadata/org.flywaydb/flyway-sqlserver/index.json
+++ b/metadata/org.flywaydb/flyway-sqlserver/index.json
@@ -1,6 +1,23 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "12.3.0",
+    "test-version" : "11.8.1",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/flywaydb/flyway-sqlserver/$version$/flyway-sqlserver-$version$-sources.jar",
+    "repository-url" : "https://github.com/flyway/flyway",
+    "test-code-url" : "N/A",
+    "documentation-url" : "https://github.com/flyway/flyway/blob/flyway-$version$/documentation/Reference/Database%20Driver%20Reference/SQL%20Server%20Database.md",
+    "description" : "Flyway SQL Server support module adds Microsoft SQL Server, Azure Synapse, and Microsoft Fabric Data Warehouse database support to Flyway. It provides the database type, parser, schema handling, and clean-mode integrations used to run Flyway migrations against those platforms.",
+    "tested-versions" : [
+      "12.3.0"
+    ],
+    "allowed-packages" : [
+      "org.flywaydb",
+      "org.flywaydb.clean",
+      "org.flywaydb.database.sqlserver"
+    ]
+  },
+  {
     "metadata-version" : "11.12.0",
     "test-version" : "11.8.1",
     "source-code-url" : "https://repo1.maven.org/maven2/org/flywaydb/flyway-sqlserver/$version$/flyway-sqlserver-$version$-sources.jar",

--- a/stats/org.flywaydb/flyway-sqlserver/12.3.0/stats.json
+++ b/stats/org.flywaydb/flyway-sqlserver/12.3.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "12.3.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1173,
+        "missed" : 3406,
+        "ratio" : 0.256169,
+        "total" : 4579
+      },
+      "line" : {
+        "covered" : 207,
+        "missed" : 587,
+        "ratio" : 0.260705,
+        "total" : 794
+      },
+      "method" : {
+        "covered" : 76,
+        "missed" : 169,
+        "ratio" : 0.310204,
+        "total" : 245
+      }
+    }
+  } ]
+}

--- a/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build.gradle
+++ b/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build.gradle
@@ -22,6 +22,7 @@ graalvmNative {
     binaries {
         all {
             buildArgs.add("-H:+AddAllCharsets")
+            buildArgs.add("--enable-url-protocols=https")
         }
     }
     agent {

